pkg/operator/status.go

```diff
@@ -289,9 +291,100 @@ func (optr *Operator) syncUpgradeableStatus() error {
 		coStatus.Message = "One or more machine config pools are updating, please see `oc get mcp` for further details"
 	}
 
+	// don't overwrite status if updating or degraded
+	if !updating && !degraded {
+		if supported, v := optr.isKubeletSkewSupported(); !supported {
+			coStatus.Reason = "KubeletSkewUnsupported"
+			coStatus.Message = fmt.Sprintf("machine config pool contains nodes with an unsupported kubelet version skew. Please see `oc get nodes` for details and upgrade all nodes so that they have a kubelet Version of at least %v.", getMinimalSkewSupportNodeVersion(v))
+			optr.eventRecorder.Eventf(co, corev1.EventTypeNormal, "KubeletSkewUnsupported", fmt.Sprintf("machine config pool contains nodes with an unsupported kubelet version skew with kube-apiserver version %v.", v))
+		}
+	}
+
 	return optr.updateStatus(co, coStatus)
 }
 
+// isKubeletSkewSupported checks the version skew of kube-apiserver and node kubelet version.
+// Returns false is the version skew is not supported. version skew > 2 is not supported.
+func (optr *Operator) isKubeletSkewSupported() (skewSupported bool, kubeAPIServerVersion string) {
+	skewSupported = true
+	nodes, err := optr.nodeLister.List(labels.Everything())
+	if err != nil {
+		return skewSupported, ""
```

Comment:
I am confused, in case of error shouldn't we returning "skewSupported=false"?

Reply:
Possibly also return error and set error message when an error has been encountered.

Reply:
See comment https://github.com/openshift/machine-config-operator/pull/2658#discussion_r663273670. The former discussion has question about this kind of error should cause failure in other processes and the message returned from here would not be very useful for customers. So ignore the error and the skew hasn't been checked at this point, shouldn't return false.

Reply:
I think what @sinnykumari is pointing out is that the issue is that if there is an error we're saying the skew IS supported, when we don't know?

Reply:
We use this func here:
```
if supported, v := optr.isKubeletSkewSupported(); !supported 
```
but if the kubeAPIServerVersion="" is intended to indicate that we didn't actually check the skew, you'd have to look at the value of v there.

Reply:
> but if the kubeAPIServerVersion="" is intended to indicate that we didn't actually check the skew, you'd have to look at the value of v there.

Here it failed to get the node kublet version. Could you explain your idea?

Reply:
Thinking about sinny's comment we can either:
1) log in the operator logs the error and the node
and also maybe:
2) log in the status the failure to check the skew and to see the operator logs for details

@sinnykumari wdyt?

Reply:
Definitely we should do 1, also doing 2 makes sense i.e. reporting error message to MCO coStatus with error message and details on where to look for further debugging.

Reply:
> Definitely we should do 1, also doing 2 makes sense i.e. reporting error message to MCO coStatus with error message and details on where to look for further debugging.

@sinnykumari  Just to be clear, is this correct: 
- log in the operator logs is using glog.Error()
- log in the status is the coStatus.Message

Reply:
@sinnykumari PTAL. Added the error to coStatus and glog.Info() to log the current status.

Reply:
This should now be resolved in the latest commit.